django/contrib/auth/__init__.py

```diff
@@ -127,7 +130,10 @@ def login(request, user, backend=None):
     request.session[HASH_SESSION_KEY] = session_auth_hash
     if hasattr(request, 'user'):
         request.user = user
-    rotate_token(request)
```

Comment:
Leave this as it is, and make the changes inside `rotate_token()`. (There's no reason for `auth.login()` to know the details of the CSRF implementation.)

Reply:
Done